transport/ice/IceSession.cpp

```diff
@@ -1185,20 +1505,15 @@ int64_t IceSession::processTimeout(const uint64_t now)
         {
             if (candidatePair->localCandidate.transportType == ice::TransportType::TCP)
             {
-                if (_eventSink)
-                {
-                    _eventSink->onIceDiscardCandidate(this,
-                        candidatePair->localEndpoint.endpoint,
-                        candidatePair->remoteCandidate.address);
-                }
                 removeCandidatePair(candidatePair);
             }
         }
-        else if (_nomination == candidatePair && candidatePair->state != ProbeState::Succeeded)
+        else if (_nomination == candidatePair && candidatePair->nominatedInProgressNextTimedout(now) == 0)
         {
-            logger::debug("found nominated pair in state %s", _logId.c_str(), toString(candidatePair->state));
```

Comment:
After timeout of the nominated candidate be in InProgress, we then remove the nomination and change the state to CONNECTIN